galsim/sed.py

```diff
@@ -230,21 +229,7 @@ def dimensionless(self):  # for convenience
         return not self.spectral
 
     def _wavelength_intersection(self, other):
-        blue_limit = self.blue_limit
-        if other.blue_limit is not None:
-            if blue_limit is None:
-                blue_limit = other.blue_limit
-            else:
-                blue_limit = max([blue_limit, other.blue_limit])
-
-        red_limit = self.red_limit
-        if other.red_limit is not None:
-            if red_limit is None:
-                red_limit = other.red_limit
-            else:
-                red_limit = min([red_limit, other.red_limit])
-
-        return blue_limit, red_limit
+        return max([self.blue_limit, other.blue_limit]), min([self.red_limit, other.red_limit])
```

Comment:
Apparently, this function is not actually used by anything, so could just remove it.